src/shared/components/QueryBuilder/QueryBuilder.tsx

```diff
@@ -78,8 +78,8 @@ function QueryBuilderComponent(props: QueryBuilderProps) {
       return;
     }
 
-    const onTransition: StateListener<QueryBuilderContext, QueryBuilderEvent> = ({ context }, event) => {
-      if (['SELECT_SUGGESTION', 'REMOVE_FILTER'].includes(event.type) && !context.isQueryUpToDate) {
+    const onTransition: StateListener<QueryBuilderContext, QueryBuilderEvent> = ({ value, context, event }) => {
+      if (value === 'idle' && !context.isQueryUpToDate && event.type !== 'EXECUTE_QUERY') {
```

Comment:
The previous logical condition was buggy in case of editing the operator of a complete filter. E.g. switching from "=" to "in" would execute the query (applying the filters to the viz panels on the view) but, because it would happen in a state that is not the "idle" state, the context of the state machine would not be updated to reflect the changes ("activateFilters" wouldn't be called). As a result, when closing the suggestions dropdown, the filter would still be rendered as inactive.